pkg/errutils/errutils.go

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2024 Arm Limited. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package errutils
+
+import (
+	"errors"
+	"fmt"
+)
+
+const (
+	ErrInvalidFileExtension     = "unsupported file extension: %s. Supported extension(s): %s"
+	ErrInvalidCmdLineArg        = "expected only one argument specifying the input file"
+	ErrFileNotExist             = "file %s does not exist"
+	ErrNoContextFound           = "no context(s) found to process"
+	ErrBinaryNotFound           = "%s binary not found %s"
+	ErrMissingPacks             = "missing packs must be installed, rerun cbuild with the --packs option"
+	ErrPathNotFound             = "path %s not found"
+	ErrInvalidContextFormat     = "invalid context format. Expected [project][.buildType][+targetType]"
+	ErrInvalidCSolutionFileName = "invalid csolution file name format. Expected '<projectname>.csolution.yml'"
```

Comment:
We need to be careful with the term projectname. In the usage we are just printing
"<name>.csolution.yml"
I am just unclear how the filename format can be "invalid"? Or are we in fact talking about the file extension? 
csolution.yml -> there is an error for that already.

Reply:
[Addressed review comments and Revised error messages](https://github.com/Open-CMSIS-Pack/cbuild/pull/243/commits/efe32d61b8c0b2faed1818051ccdf45aba4bd29e)